lib/ramble/ramble/cmd/workspace.py

```diff
@@ -955,6 +956,134 @@ def workspace_mirror(args):
     pipeline.run()
 
 
+def workspace_generate_config_setup_parser(subparser):
+    """generate current workspace config"""
+
+    arguments.add_common_arguments(subparser, ["application"])
+
+    subparser.add_argument(
+        "--workload-filter",
+        "--wf",
+        dest="workload_filters",
+        action="append",
+        help="filter to use when selecting workloads in the application. "
+        + "Workload is kept if it matches any filter.",
+    )
+
+    subparser.add_argument(
+        "--variable-filter",
+        "--vf",
+        dest="variable_filters",
+        action="append",
+        help="filter to use when selecting variables in the workloads. "
+        + "Variable is kept if it matches any filter.",
+    )
+
+    subparser.add_argument(
+        "--variable-definition",
+        "-v",
+        dest="variable_definitions",
+        action="append",
+        help="variable definition to set in the generated experiments. "
+        + "Given in the form key=value",
+    )
+
+    subparser.add_argument(
+        "--experiment-name",
+        "-e",
+        dest="experiment_name",
+        default="generated",
+        help="name of generated experiment",
+    )
+
+    subparser.add_argument(
+        "--package-manager",
```

Comment:
I got into trouble with this in testing where I wanted to update the package manager for a config I had just generated. I think we should either (or maybe both):

1. Warn when a generated experiment would collide so we refuse to emit it; and
2. Perhaps allow a `--force` option that has a semantic like "I know this will collide and overwrite config, but I'm sure I want it"

Reply:
Done!